### PR TITLE
Fix tests on macOS+ refactoring

### LINKED ISSUE #10651: Test failures on macOS
I got these errors when I did a clean checkout then `gh pr checkout 10650` followed by `./gradlew jacocoTestReport --rerun-tasks` on my macOS machine, they look totally unrelated. Need to investigate them separately, they clearly work in CI and they must work on your M1 machine David, 🤷 

I don't have any build reasons to stop this

```
com.ichi2.anki.jsaddons.TgzPackageExtractTest > extractTarGzipToAddonFolderTest FAILED
    java.io.IOException: There is not enough space to import the package.
    Needed - 1.45 kB 
    Available - 0 B.

    Having Trouble? Try to free space on your phone if possible. Alternatively, sync uses less space than import. You may try importing the package with a device with more space (for example Anki Desktop) and then sync it to this device.
        at com.ichi2.anki.jsaddons.TgzPackageExtract.extractTarGzipToAddonFolder(TgzPackageExtract.kt:123)
        at com.ichi2.anki.jsaddons.TgzPackageExtractTest.extractTarGzipToAddonFolderTest(TgzPackageExtractTest.kt:82)

com.ichi2.anki.jsaddons.TgzPackageExtractTest > unGzipTest FAILED
    org.apache.commons.compress.archivers.ArchiveException: File extract exceeds storage space
        at com.ichi2.anki.jsaddons.TgzPackageExtract.enforceSpaceUsedLessThanHalfAvailable(TgzPackageExtract.kt:360)
        at com.ichi2.anki.jsaddons.TgzPackageExtract.unGzip(TgzPackageExtract.kt:180)
        at com.ichi2.anki.jsaddons.TgzPackageExtractTest.unGzipTest(TgzPackageExtractTest.kt:143)

com.ichi2.anki.jsaddons.TgzPackageExtractTest > unTarTest FAILED
    org.apache.commons.compress.archivers.ArchiveException: File extract exceeds storage space
        at com.ichi2.anki.jsaddons.TgzPackageExtract.enforceSpaceUsedLessThanHalfAvailable(TgzPackageExtract.kt:360)
        at com.ichi2.anki.jsaddons.TgzPackageExtract.unGzip(TgzPackageExtract.kt:180)
        at com.ichi2.anki.jsaddons.TgzPackageExtractTest.unTarTest(TgzPackageExtractTest.kt:112)

```

_Originally posted by @mikehardy in https://github.com/ankidroid/Anki-Android/issues/10650#issuecomment-1082304654_

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
`IOException: There is not enough space to import the package.`

Cause: `TgzPackageExtractTest` - `ShadowStatsFs` fails to use the canonical path

But: The non-canonical path is required by `BackupManagerTestUtilities`

## Fixes
Fixes #10651

## Approach
* define a method which uses both canonical and regular paths `registerStats`

## How Has This Been Tested?

Ran tests - now pass

## Learning (optional, can help others)
* Macs are hard yo, but it's nice to be able to reasonably run tests locally
* ShadowStatFs needs BOTH canonical and non-canonical path 

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/anki/jsaddons/TgzPackageExtractTest.kt [hunk 1]
```diff
@@ -39,19 +40,23 @@ class TgzPackageExtractTest : RobolectricTest() {
     override fun setUp() {
         super.setUp()
 
-        var currentAnkiDroidDirectory = CollectionHelper.getCurrentAnkiDroidDirectory(targetContext)
-        val path = File(currentAnkiDroidDirectory, "addons").parentFile?.path
-        ShadowStatFs.registerStats(path, 100, 20, 10000)
+        val currentAnkiDroidDirectory = CollectionHelper.getCurrentAnkiDroidDirectory(targetContext)
+        addonDir = File(currentAnkiDroidDirectory, "addons")
+        ShadowStatFs.markAsNonEmpty(addonDir.parentFile!!)
```
[the remark this answers is not in the document]
Comment (krmanik) [reply AnkiDroid/src/test/java/com/ichi2/anki/jsaddons/TgzPackageExtractTest.kt:45]: It should be `currentAnkiDroidDirectory`.

```
AnkiDroid/addons/some-addons-folder
```

**Review by krmanik — APPROVED** [review]

**Comment by mikehardy** [comment]
You keep saying "apple silicon" but these are generic macOS problems ;-)

File: AnkiDroid/src/test/java/com/ichi2/testutils/ShadowStatFs.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * @param blockCount number of blocks
+     * @param freeBlocks number of free blocks
+     * @param availableBlocks number of available blocks
+     *
+     * @see [markAsNonEmpty]
+     */
+    @JvmStatic
+    fun registerStats(path: File, blockCount: Int, freeBlocks: Int, availableBlocks: Int) {
+        RobolectricStats.registerStats(path, blockCount, freeBlocks, availableBlocks)
+        // call canonicalFile so this works on M1 Macs
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/testutils/ShadowStatFs.kt:42]:
```suggestion
        // call canonicalFile so this works on Macs
```

**Review by mikehardy — APPROVED** [review]
All LGTM - tiniest nit is the non-M1-specificness.
Argues for doing a unit test run on windows and macOS as a matrix really to make sure our dev environments run out of the box, we've started taking some entropy there and used to test them on travis. I'll add an issue

**Comment by mikehardy** [comment]
Just the comment change then mash that green button!

**Pull request merged**